### PR TITLE
Add table filtering

### PR DESCRIPTION
**Pull request by konradszydlo**
Table filtering/searching based on a string from input.

TODO:
- more pleasant styling
- adjusting pagination based on filtered rows

### CODE REVIEW

File: src/re_frame_datatable/filtering.cljc [hunk 1]
```diff
@@ -0,0 +1,32 @@
+(ns re-frame-datatable.filtering
+  (:require [re-frame.core :as re-frame :refer [trim-v]]
+            [re-frame-datatable.db :as db]
+            [re-frame-datatable.subs :as subs]))
+
+(defn- set-filter-term [db id term v]
+  (db/update-state db id assoc-in [:filtering term] v))
+
+;;; TODO adjust pagination based on filtering
```
Comment (wneirynck) [inline src/re_frame_datatable/filtering.cljc:9]: Isn't that already being taken care of in the `:data` sub?

File: src/re_frame_datatable/subs.cljc [hunk 2]
```diff
@@ -15,6 +15,23 @@
  (fn [db [_ db-id]]
    (db/options db db-id)))
 
+(defn- apply-case-sensitivity
+  "If case-insensitivity param is set to true lowercase passed string"
+  [s case-insensitive]
+  (if case-insensitive
+    (clojure.string/lower-case s)
```
Comment (wneirynck) [inline src/re_frame_datatable/subs.cljc]: You could make this more generic by adding a `:filter-fn` or `:filter-tx` configuration parameter that allows you to specify the filter function/transducer.

**Comment by wneirynck** [comment]
Could you also update documentation to explain how filtering works?

**Comment by konradszydlo** [comment]
New commit with updated documentation pushed. New filtering section added to the docs project.

File: src/re_frame_datatable/subs.cljc (hunk 2 above)
Comment (konradszydlo) [reply src/re_frame_datatable/subs.cljc]: filtering-fn added as a option for filtering. Usage and default behaviour explained in a subsection of filtering in docs. Defaults to case sensitive filtering.

**Pull request merged**